app/services/mhv_logging_service.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+require 'mhv_logging/client'
+class MHVLoggingService
+  def self.login(current_user)
+    # If login has already been submitted, do nothing
+    return false if current_user.mhv_last_signed_in || current_user.mhv_correlation_id.nil?
```

Comment:
for login that extra || can actually be removed. I left it because it is symmetrical to the logout one to have it in there, but the logout one definitely needs it, and yes, it makes sense maybe to check for the correlation_id first. I'll make this change.